kv/kvprober: TestProberDoesReadsAndWrites failed

kv/kvprober.TestProberDoesReadsAndWrites [failed](https://mesolite.cluster.engflow.com/invocations/default/133048a9-c08d-4d0e-9e47-ea2272d30fec?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3Zwcm9iZXI6a3Zwcm9iZXJfdGVzdA==) on master @ [ac03277493425190db09f9c2229689ddc51047f4](https://github.com/cockroachdb/cockroach/commits/ac03277493425190db09f9c2229689ddc51047f4):


```
=== RUN   TestProberDoesReadsAndWrites
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestProberDoesReadsAndWrites3167472009
    test_log_scope.go:76: use -show-logs to present logs inline
    kvprober_integration_test.go:261: -- test log scope end --
--- FAIL: TestProberDoesReadsAndWrites (31.41s)
=== RUN   TestProberDoesReadsAndWrites/happy_path
    kvprober_integration_test.go:119: 
        	Error Trace:	pkg/kv/kvprober/kvprober_integration_test.go:119
        	Error:      	Should be zero, but was 1
        	Test:       	TestProberDoesReadsAndWrites/happy_path
    --- FAIL: TestProberDoesReadsAndWrites/happy_path (7.63s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProberDoesReadsAndWrites.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45218

kv/kvprober.TestProberDoesReadsAndWrites [failed](https://tanzanite.cluster.engflow.com/invocations/default/a3b78e3b-0ad5-495b-ad9a-a9e70ce6adfa?testReportRun=19&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cva3Yva3Zwcm9iZXI6a3Zwcm9iZXJfdGVzdA==) on master @ [fa9acaa8a8b8b58b1448c61ca5a8f6e6a992d64c](https://github.com/cockroachdb/cockroach/commits/fa9acaa8a8b8b58b1448c61ca5a8f6e6a992d64c):


```
=== RUN   TestProberDoesReadsAndWrites
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestProberDoesReadsAndWrites2704992279
    test_log_scope.go:76: use -show-logs to present logs inline
    kvprober_integration_test.go:261: -- test log scope end --
test logs left over in: outputs.zip/logTestProberDoesReadsAndWrites2704992279
--- FAIL: TestProberDoesReadsAndWrites (29.48s)
=== RUN   TestProberDoesReadsAndWrites/all_ranges_are_unavailable_for_Puts_only
    kvprober_integration_test.go:257: 
        	Error Trace:	pkg/kv/kvprober/kvprober_integration_test.go:257
        	Error:      	Should be zero, but was 1
        	Test:       	TestProberDoesReadsAndWrites/all_ranges_are_unavailable_for_Puts_only
    --- FAIL: TestProberDoesReadsAndWrites/all_ranges_are_unavailable_for_Puts_only (2.60s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=19</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestProberDoesReadsAndWrites.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvprober/BUILD.bazel
+++ b/pkg/kv/kvprober/BUILD.bazel
@@ -48,6 +48,7 @@ go_test(
         "//pkg/kv",
         "//pkg/kv/kvpb",
         "//pkg/kv/kvserver",
+        "//pkg/kv/kvserver/storeliveness/storelivenesspb",
         "//pkg/roachpb",
         "//pkg/security/securityassets",
         "//pkg/security/securitytest",

--- a/pkg/kv/kvprober/kvprober_integration_test.go
+++ b/pkg/kv/kvprober/kvprober_integration_test.go
@@ -22,6 +22,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvprober"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver"
+	slpb "github.com/cockroachdb/cockroach/pkg/kv/kvserver/storeliveness/storelivenesspb"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
@@ -411,9 +412,29 @@ func initTestServer(
 	s, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{
 		// KV probes always go to the storage layer.
 		DefaultTestTenant: base.TestIsSpecificToStorageLayerAndNeedsASystemTenant,
+		Settings:          cluster.MakeClusterSettings(),
+		Knobs:             knobs,
+		RaftConfig: base.RaftConfig{
+			// Speed up tests.
+			RaftTickInterval:           100 * time.Millisecond,
+			RaftElectionTimeoutTicks:   2,
+			RaftHeartbeatIntervalTicks: 1,
+		},
+	})
 
-		Settings: cluster.MakeClusterSettings(),
-		Knobs:    knobs,
+	// With leader leases, wait for store liveness support to be established. This
+	// is useful as the leader will only be elected once it has quorum support
+	// in store liveness. Without this, we could see some deadline exceeded errors
+	// until the store liveness support is established and the leader is elected.
+	store, err := s.GetStores().(*kvserver.Stores).GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
+	testutils.SucceedsSoon(t, func() error {
+		ident := slpb.StoreIdent{NodeID: store.NodeID(), StoreID: store.StoreID()}
+		epoch, _ := store.TestingStoreLivenessSupportManager().SupportFrom(ident)
+		if epoch == 0 {
+			return errors.New("support not established")
+		}
+		return nil
 	})
 
 	// Given small test cluster, this better exercises the planning logic.

